Implemented intraday and historical stock price endpoints

avidlaud opened the pull request:
Created the stock service endpoints for requesting prices. Addresses #15

# Endpoints
## Intraday:
`{url}/equity/intraday/:equityTicker`

Usage:
Intraday prices for Twitter (twtr):
`{url}/equity/intraday/twtr`

Return body:
Object of time:average_price pairs
![image](https://user-images.githubusercontent.com/31025257/111838414-b28b1800-88cf-11eb-8c79-cfc5a28f081e.png)


## Historical:
`{url}/equity/historical/:equityTicker/?timeframe=`
Allows arguments of: `'5d', '1m', '6m', 'ytd', '1y',  or '5y'`

Usage:
1 month prices for Twitter (twtr):
`{url}/equity/historical/twtr/?timeframe=1m`

Return body:
Object of date:close_price pairs

![image](https://user-images.githubusercontent.com/31025257/111838571-e6663d80-88cf-11eb-9f1d-2169be63e511.png)

# Caching
The variable `CACHE_TIME` in `stockController.js` defines the amount of time that an entry is considered recent enough. The value is in minutes.

JScheire commented on app/services/stock/controllers/stockController.js:
```diff
@@ -0,0 +1,101 @@
+        );
+        return updatedEquity.intraday;
+    } catch (err) {
+        console.error(err);
```
can we send as a response with appropriate status code

JScheire commented on app/services/stock/controllers/stockController.js:
```diff
@@ -0,0 +1,101 @@
+        );
+        return updatedEquity[timeFrame];
+    } catch (err) {
+        console.error(err);
```
same thing here

avidlaud commented:
I'll address the error handling issues tomorrow; will also implement a `/equity/current/:ticker` endpoint for easier order and portfolio calculations.

avidlaud commented:
Added some basic error handling - it's not very descriptive but all calls are requests are made internally in our application so it shouldn't be much of an issue.

Implemented `/equity/current/:ticker` endpoint for easy current price access.

avidlaud commented:
I think this should now cleanly merge
